int8 count
body[6] bodies